### PR TITLE
download OCP nightly through GA'ed version incase of failures

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu [vavuthu@redhat.com](mailto:vavuthu@redhat.com)

### CODE REVIEW

**Comment by vavuthu** [comment]
due to resource constarint, job is in queue, in the meantime verified here ( success downloaded nightly version via GA'ed ):  https://url.corp.redhat.com/0e0c3e7

```
2024-06-26 17:55:02  08:25:02 - MainThread - ocs_ci.utility.utils - INFO  - Downloading openshift client (4.16.0-0.nightly-2024-06-25-194629).
2024-06-26 17:55:02  08:25:02 - MainThread - ocs_ci.utility.utils - INFO  - Directory '/home/jenkins/bin' successfully created.
2024-06-26 17:55:02  08:25:02 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ldd --version ldd
2024-06-26 17:55:02  08:25:02 - MainThread - ocs_ci.utility.utils - ERROR  - Failed to download the url https://xxxxx/4.16.0-0.nightly-2024-06-25-194629/openshift-client-linux-amd64-rhel9-4.16.0-0.nightly-2024-06-25-194629.tar.gz with exception 'The URL https://xxxxx/4.16.0-0.nightly-2024-06-25-194629/openshift-client-linux-amd64-rhel9-4.16.0-0.nightly-2024-06-25-194629.tar.gz is not available! Status: 404.'
2024-06-26 17:55:03  08:25:03 - MainThread - ocs_ci.utility.utils - INFO  - version 4.15 is GA'ed, use the same version to download oc
2024-06-26 17:55:03  08:25:03 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ldd --version ldd
2024-06-26 17:55:04  08:25:04 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: tar xzvf openshift-client.tar.gz -C /tmp
2024-06-26 17:55:06  08:25:06 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ldd --version ldd
2024-06-26 17:55:06  08:25:06 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: /tmp/oc adm release extract -a /<path>/pull-secret --command=oc registry.ci.openshift.org/ocp/release:4.16.0-0.nightly-2024-06-25-194629 --to .
2024-06-26 17:55:13  08:25:12 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: /home/jenkins/bin/oc version --client
2024-06-26 17:55:13  08:25:12 - MainThread - ocs_ci.utility.utils - INFO  - OpenShift Client version: Client Version: 4.16.0-0.nightly-2024-06-25-194629
2024-06-26 17:55:13  Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3

```

**Review by petr-balogh — APPROVED** [review]

File: ocs_ci/utility/version.py [hunk 1]
```diff
@@ -152,3 +152,22 @@ def compare_versions(expression):
...
+        version (str): Version ( eg: 4.16, 4.16.0-0.nightly-2024-06-25-194629)
+        count (int): previous version count. if count is 1, it will get 1st previous version.
+            if count is 2, it will get 2nd previous version.
+
+    Returns:
+        str: Previous version ( returns only major and minor version, eg: 4.15 )
+
+    """
+    version = get_semantic_version(version, only_major_minor=True)
+    new_minor = version.minor - count
```
Comment (dahorak) [inline ocs_ci/utility/version.py:171]: There will be an issue when the major version will change (e.g. with `5.0`) - but that is not relevant for now.

**Review by dahorak — APPROVED** [review]
LGTM

File: ocs_ci/utility/version.py (hunk 1 above)
Comment (vavuthu) [reply ocs_ci/utility/version.py:171]: even I had this scenario in mind while writing and don't know how to handle that. will handle later

**Pull request merged**